### PR TITLE
Add Single-Host v5e and v6e RayService Samples for vLLM Guide

### PR DESCRIPTION
**Pull request by ryanaoleary**
## Description

This PR adds RayCluster and RayService samples for both v5e and v6e single-host TPUs. These samples contain some placeholder values beginning with `$` which will be replaced with environment variables previously set in the guide when running `envsubst < ray-cluster.tpu-v5e-singlehost.yaml | kubectl apply -f -`. These samples will be used in a GKE guide showcasing vLLM and Ray Serve with single-host TPUs.

TODO: This PR should be merged prior to the multi-host v5e/v6e counterpart to this PR (https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/1467) since it'll use some of the same scripts.

## Tasks

* [x] The [contributing guide](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/.github/CONTRIBUTING.md) has been read and followed.
* [x] The samples added / modified have been fully tested.
* [x] Workflow files have been added / modified, if applicable.
* [x] Region tags have been properly added, if new samples.
* [x] All dependencies are set to up-to-date versions, as applicable.
* [ ] Merge this pull-request for me once it is approved.

### CODE REVIEW

**Comment by ryanaoleary** [comment]
@bourgeoisor pinging another possible reviewer because I believe @alizaidis is OOO.

**Review by NimJay — APPROVED** [review]
This is amazing! Thanks, @ryanaoleary!

Looks good.
I assume you've tested this appropriately.
Approved. Let me know if I can go ahead and merge this.

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+          annotations:
+            gke-gcsfuse/volumes: "true"
+            gke-gcsfuse/cpu-limit: "0"
+            gke-gcsfuse/memory-limit: "0"
+            gke-gcsfuse/ephemeral-storage-limit: "0"
+        spec:
+          serviceAccountName: $KSA_NAME
+          containers:
+          - name: ray-head
+            image: $VLLM_IMAGE
```
Comment (NimJay) [inline ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml:50]: **Question:** What will `$VLLM_IMAGE` be set to? Curious which image the tutorial will be using.

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            emptyDir:
+              medium: Memory
+          - name: gcs-fuse-csi-ephemeral
+            csi:
+              driver: gcsfuse.csi.storage.gke.io
+              volumeAttributes:
+                bucketName: $GSBUCKET
+                mountOptions: "implicit-dirs,file-cache:enable-parallel-downloads:true,file-cache:parallel-downloads-per-file:100,file-cache:max-parallel-downloads:-1,file-cache:download-chunk-size-mb:10,file-cache:max-size-mb:-1"
+    workerGroupSpecs:
+    - replicas: 2
```
Comment (NimJay) [inline ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml:94]: **Question:** Was this intentional? `replicas` is 2, but `maxReplicas` is 1.

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                google.com/tpu: "8"
+                ephemeral-storage: 50Gi
+                memory: 100G
+              requests:
+                cpu: "100"
+                google.com/tpu: "8"
+                ephemeral-storage: 50Gi
+                memory: 100G
+                        securityContext:
+              privileged: true
```
Comment (NimJay) [inline ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml:130]: **Issue:** Would this formatting issue cause trouble?
```
                memory: 100G
                        securityContext:
              privileged: true
```

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v6e-singlehost.yaml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            emptyDir:
+              medium: Memory
+          - name: gcs-fuse-csi-ephemeral
+            csi:
+              driver: gcsfuse.csi.storage.gke.io
+              volumeAttributes:
+                bucketName: $GSBUCKET
+                mountOptions: "implicit-dirs,file-cache:enable-parallel-downloads:true,file-cache:parallel-downloads-per-file:100,file-cache:max-parallel-downloads:-1,file-cache:download-chunk-size-mb:10,file-cache:max-size-mb:-1"
+    workerGroupSpecs:
+    - replicas: 2
```
Comment (NimJay) [inline ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v6e-singlehost.yaml:94]: **Question:** `replicas: 2` while `maxReplicas: 1`. Was this intentional?

File: ai-ml/gke-ray/rayserve/llm/tpu/ray-cluster.tpu-v5e-singlehost.yaml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            imagePullPolicy: IfNotPresent
+            resources:
+              limits:
+                cpu: "8"
+                ephemeral-storage: 5Gi
+                memory: 40G
+              requests:
+                cpu: "8"
+                ephemeral-storage: 5Gi
+                memory: 40G
```
Comment (NimJay) [inline ai-ml/gke-ray/rayserve/llm/tpu/ray-cluster.tpu-v5e-singlehost.yaml:44]: **Question:** Is `memory: 40G` and `cpu: 8` necessary? Are we able to use less resources (to make this tutorial more approachable and less expensive)?

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml (hunk 1 above)
Comment (ryanaoleary) [reply ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml:50]: A vLLM image built with these instructions: https://docs.vllm.ai/en/v0.5.5/getting_started/tpu-installation.html from the `Dockerfile.tpu` file. This commit has the updated requirements and Dockerfile: https://github.com/vllm-project/vllm/commit/cd34029e91ad2d38a58d190331a65f9096c0b157

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml (hunk 3 above)
Comment (ryanaoleary) [reply ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml]: Oh we don't need those lines anymore, I'll went ahead and removed them in 
c87b364. Setting `securityContext: privileged: true` was necessary due to a bug in an older version of tpu_device_plugin, but is fixed in the minimum cluster version we require in the guide.

File: ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml (hunk 2 above)
Comment (ryanaoleary) [reply ai-ml/gke-ray/rayserve/llm/model-composition/ray-service.tpu-v5e-singlehost.yaml:92]: Fixed in c87b364

File: ai-ml/gke-ray/rayserve/llm/tpu/ray-cluster.tpu-v5e-singlehost.yaml (hunk 5 above)
Comment (ryanaoleary) [reply ai-ml/gke-ray/rayserve/llm/tpu/ray-cluster.tpu-v5e-singlehost.yaml:44]: We're not installing any additional dependencies not included in the image, so I think these can probably be lowered. I'll try it with `cpu: "2"` and `memory: "8G"` to validate this works first.

Comment (ryanaoleary) [reply ai-ml/gke-ray/rayserve/llm/tpu/ray-cluster.tpu-v5e-singlehost.yaml]: Sorry for the delay @NimJay, I was finally able to scale up some slices and test with the lower resource requests. Everything works well from my manual testing so this PR should be good to merge whenever is convenient.

**Pull request merged**